cpp/docs/DEVELOPER_GUIDE.md

```diff
@@ -413,7 +413,7 @@ should avoid creating streams (even if it is slightly less efficient). It is a g
 
 ## Memory Allocation
 
-Device [memory resources](#memory_resource) are used in libcudf to abstract and control how device
+Device [memory resources](#rmmdevice_memory_resource) are used in libcudf to abstract and control how device
```

Comment:
Where is this pointing to?

Reply:
This points to this section in the same file: https://github.com/rapidsai/cudf/blob/d6269e28051866f47dad5825edc57cd5392a7969/cpp/docs/DEVELOPER_GUIDE.md#rmmdevice_memory_resource
It looks like the anchor link is specified as: `memory_resource` according to the source
```
## `rmm::device_memory_resource`<a name="memory_resource"></a>

libcudf allocates all device memory via RMM memory resources (MR). See the
[RMM documentation](https://github.com/rapidsai/rmm/blob/main/README.md) for details.
```
Perhaps the link is not working.

Reply:
link is working. `xrefcheck` tool did not understand the inline html code. Thought this change is not required, this change is made in preparation for adding `xrefcheck` tool as a CI check eventually to avoid broken links in our documentation.

Reply:
Should the inline html code be removed?

Reply:
I originally searched for `rmmdevice_memory_resource` and couldn't find it, which is why I was confused. How is the xref name constructed such that name="memory_resource" is getting translated to `rmmdevice_memory_resource`?

Reply:
It can be removed. `rmmdevice_memory_resource` is constructed from ``## `rmm::device_memory_resource` ``